src/transformers/models/deprecated/open_llama/modeling_open_llama.py

```diff
@@ -161,23 +161,37 @@ def __init__(self, dim, max_position_embeddings=2048, base=10000, device=None, s
         super().__init__(dim, max_position_embeddings, base, device)
 
     def _set_cos_sin_cache(self, seq_len, device, dtype):
+        if seq_len <= self.max_position_embeddings:
+            seq_len = self.max_position_embeddings
+
```

Comment:
@gante `seq_len` is set to `max_position_embeddings` here. Minimal cached length is `max_position_embeddings`. I think `seq_len <= self._seq_len_cached==self.max_position_embeddings` works.

Extra time mainly caused by recompute `max(seq_len - max_position_embeddings, 0) * len(generated_token)` times. Haven't thought of a good way yet.

Reply:
I see, you're right! I forgot to account for this change :)